features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java

```diff
@@ -120,4 +162,9 @@ public void setStateManager(StateManager stateManager) {
     public void setUsageStatisticsReporter(UsageStatisticsReporter usageStatisticsReporter) {
         m_usageStatisticsReporter = Objects.requireNonNull(usageStatisticsReporter);
     }
+
+    private WebApplicationException getException(String msg, Throwable e) {
```

Comment:
I think you can omit `WebApplicationException` and directly return Response.  Is there any advantage of using `WebApplicationException` ?

Reply:
Not really sure ... just following pattern in many other v2 REST services. I'll change to just return Response and do the logging.

Reply:
Updated to log exception and return `Response`.